fairlearn/metrics/_metric_frame.py

```diff
@@ -272,26 +290,39 @@ def __init__(self,
                  y_true,
                  y_pred,
                  sensitive_features,
-                 control_features: Optional = None,
+                 control_features=None,
                  sample_params: Optional[Union[Dict[str, Any], Dict[str, Dict[str, Any]]]] = None):
         """Read a placeholder comment."""
         check_consistent_length(y_true, y_pred)
+
+        all_data = pd.DataFrame(dtype='object')
+
         y_t = _convert_to_ndarray_and_squeeze(y_true)
         y_p = _convert_to_ndarray_and_squeeze(y_pred)
 
-        func_dict = self._process_functions(metrics, sample_params)
+        all_data['y_true'] = list(y_t)
```

Comment:
why do we convert things to an ndarray and then a list here?

Reply:
Because if you try assigning a 2D array to a single pandas column, it complains. However, it doesn't object to a list-of-lists.